### ISSUE TITLE
'float16x4_t' was not declared in this scope when compile for hisi

### ISSUE THREAD

**Issue by live106** [Tencent/ncnn#4433]
## error log | 日志或报错信息 | ログ
[  0%] Built target ncnn-generate-spirv
[  0%] Building CXX object src/CMakeFiles/ncnn.dir/blob.cpp.o
[  1%] Building CXX object src/CMakeFiles/ncnn.dir/allocator.cpp.o
[  1%] Building CXX object src/CMakeFiles/ncnn.dir/benchmark.cpp.o
[  2%] Building CXX object src/CMakeFiles/ncnn.dir/command.cpp.o
[  2%] Building CXX object src/CMakeFiles/ncnn.dir/cpu.cpp.o
[  4%] Building CXX object src/CMakeFiles/ncnn.dir/datareader.cpp.o
[  4%] Building CXX object src/CMakeFiles/ncnn.dir/c_api.cpp.o
[  4%] Building CXX object src/CMakeFiles/ncnn.dir/gpu.cpp.o
[  4%] Building CXX object src/CMakeFiles/ncnn.dir/mat.cpp.o
[  5%] Building CXX object src/CMakeFiles/ncnn.dir/layer.cpp.o
[  6%] Building CXX object src/CMakeFiles/ncnn.dir/mat_pixel_drawing.cpp.o
[  7%] Building CXX object src/CMakeFiles/ncnn.dir/mat_pixel_affine.cpp.o
[  7%] Building CXX object src/CMakeFiles/ncnn.dir/mat_pixel.cpp.o
[  8%] Building CXX object src/CMakeFiles/ncnn.dir/mat_pixel_resize.cpp.o
[  8%] Building CXX object src/CMakeFiles/ncnn.dir/mat_pixel_rotate.cpp.o
[  9%] Building CXX object src/CMakeFiles/ncnn.dir/modelbin.cpp.o
[ 10%] Building CXX object src/CMakeFiles/ncnn.dir/net.cpp.o
[ 10%] Building CXX object src/CMakeFiles/ncnn.dir/option.cpp.o
[ 10%] Building CXX object src/CMakeFiles/ncnn.dir/pipeline.cpp.o
[ 11%] Building CXX object src/CMakeFiles/ncnn.dir/paramdict.cpp.o
[ 12%] Building CXX object src/CMakeFiles/ncnn.dir/pipelinecache.cpp.o
[ 12%] Building CXX object src/CMakeFiles/ncnn.dir/simpleocv.cpp.o
[ 14%] Building CXX object src/CMakeFiles/ncnn.dir/simpleomp.cpp.o
[ 14%] Building CXX object src/CMakeFiles/ncnn.dir/simplestl.cpp.o
[ 14%] Building CXX object src/CMakeFiles/ncnn.dir/layer/absval.cpp.o
[ 15%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/absval_arm.cpp.o
[ 15%] Building CXX object src/CMakeFiles/ncnn.dir/layer/batchnorm.cpp.o
[ 16%] Building CXX object src/CMakeFiles/ncnn.dir/layer/bias.cpp.o
[ 17%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/batchnorm_arm.cpp.o
[ 17%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/bias_arm.cpp.o
[ 17%] Building CXX object src/CMakeFiles/ncnn.dir/layer/concat.cpp.o
[ 18%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/concat_arm.cpp.o
[ 19%] Building CXX object src/CMakeFiles/ncnn.dir/layer/bnll.cpp.o
[ 19%] Building CXX object src/CMakeFiles/ncnn.dir/layer/convolution.cpp.o
[ 20%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/convolution_arm.cpp.o
[ 21%] Building CXX object src/CMakeFiles/ncnn.dir/layer/crop.cpp.o
[ 21%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/crop_arm.cpp.o
[ 22%] Building CXX object src/CMakeFiles/ncnn.dir/layer/deconvolution.cpp.o
[ 22%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/deconvolution_arm.cpp.o
[ 23%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/dropout_arm.cpp.o
[ 23%] Building CXX object src/CMakeFiles/ncnn.dir/layer/eltwise.cpp.o
[ 24%] Building CXX object src/CMakeFiles/ncnn.dir/layer/dropout.cpp.o
[ 24%] Building CXX object src/CMakeFiles/ncnn.dir/layer/elu.cpp.o
[ 25%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/eltwise_arm.cpp.o
[ 25%] Building CXX object src/CMakeFiles/ncnn.dir/layer/exp.cpp.o
[ 27%] Building CXX object src/CMakeFiles/ncnn.dir/layer/flatten.cpp.o
[ 27%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/flatten_arm.cpp.o
[ 28%] Building CXX object src/CMakeFiles/ncnn.dir/layer/embed.cpp.o
[ 29%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/innerproduct_arm.cpp.o
[ 29%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/innerproduct_arm_vfpv4.cpp.o
[ 29%] Building CXX object src/CMakeFiles/ncnn.dir/layer/innerproduct.cpp.o
[ 30%] Building CXX object src/CMakeFiles/ncnn.dir/layer/input.cpp.o
[ 30%] Building CXX object src/CMakeFiles/ncnn.dir/layer/log.cpp.o
[ 31%] Building CXX object src/CMakeFiles/ncnn.dir/layer/lrn.cpp.o
[ 32%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/lrn_arm.cpp.o
[ 32%] Building CXX object src/CMakeFiles/ncnn.dir/layer/memorydata.cpp.o
[ 33%] Building CXX object src/CMakeFiles/ncnn.dir/layer/mvn.cpp.o
[ 35%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/pooling_arm.cpp.o
[ 35%] Building CXX object src/CMakeFiles/ncnn.dir/layer/power.cpp.o
[ 35%] Building CXX object src/CMakeFiles/ncnn.dir/layer/prelu.cpp.o
[ 36%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/prelu_arm.cpp.o
[ 36%] Building CXX object src/CMakeFiles/ncnn.dir/layer/pooling.cpp.o
[ 37%] Building CXX object src/CMakeFiles/ncnn.dir/layer/reduction.cpp.o
[ 37%] Building CXX object src/CMakeFiles/ncnn.dir/layer/proposal.cpp.o
[ 37%] Building CXX object src/CMakeFiles/ncnn.dir/layer/relu.cpp.o
[ 38%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/relu_arm.cpp.o
[ 39%] Building CXX object src/CMakeFiles/ncnn.dir/layer/reshape.cpp.o
[ 39%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/reshape_arm.cpp.o
[ 40%] Building CXX object src/CMakeFiles/ncnn.dir/layer/roipooling.cpp.o
[ 40%] Building CXX object src/CMakeFiles/ncnn.dir/layer/scale.cpp.o
[ 41%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/scale_arm.cpp.o
[ 42%] Building CXX object src/CMakeFiles/ncnn.dir/layer/sigmoid.cpp.o
In file included from /home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_arm.cpp:31:0:
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_fp16s.h: In function 'void ncnn::innerproduct_pack4_fp16s_neon(const ncnn::Mat&, ncnn::Mat&, const ncnn::Mat&, const ncnn::Mat&, int, const ncnn::Mat&, const ncnn::Option&)':
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_fp16s.h:256:45: error: 'float16x4_t' was not declared in this scope
             float32x4_t _w0 = vcvt_f32_f16((float16x4_t)(vget_low_u16(_w01)));
                                             ^~~~~~~~~~~
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_fp16s.h:256:77: error: 'vcvt_f32_f16' was not declared in this scope
             float32x4_t _w0 = vcvt_f32_f16((float16x4_t)(vget_low_u16(_w01)));
                                                                             ^
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_fp16s.h:284:44: error: 'float16x4_t' was not declared in this scope
             float32x4_t _w = vcvt_f32_f16((float16x4_t)(vld1_u16(kptr)));
                                            ^~~~~~~~~~~
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_fp16s.h:284:72: error: 'vcvt_f32_f16' was not declared in this scope
             float32x4_t _w = vcvt_f32_f16((float16x4_t)(vld1_u16(kptr)));
                                                                        ^
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_fp16s.h: In function 'void ncnn::innerproduct_fp16s_neon(const ncnn::Mat&, ncnn::Mat&, const ncnn::Mat&, const ncnn::Mat&, int, const ncnn::Mat&, const ncnn::Option&)':
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_fp16s.h:413:45: error: 'float16x4_t' was not declared in this scope
             float32x4_t _w0 = vcvt_f32_f16((float16x4_t)(vld1_u16(kptr0)));
                                             ^~~~~~~~~~~
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_fp16s.h:413:74: error: 'vcvt_f32_f16' was not declared in this scope
             float32x4_t _w0 = vcvt_f32_f16((float16x4_t)(vld1_u16(kptr0)));
                                                                          ^
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_fp16s.h:510:44: error: 'float16x4_t' was not declared in this scope
             float32x4_t _w = vcvt_f32_f16((float16x4_t)(vld1_u16(kptr)));
                                            ^~~~~~~~~~~
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_fp16s.h:510:72: error: 'vcvt_f32_f16' was not declared in this scope
             float32x4_t _w = vcvt_f32_f16((float16x4_t)(vld1_u16(kptr)));
                                                                        ^
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_fp16s.h: In function 'void ncnn::innerproduct_transform_kernel_fp16s_neon(const ncnn::Mat&, ncnn::Mat&, int, int, const ncnn::Option&)':
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_fp16s.h:716:68: error: 'vcvt_f16_f32' was not declared in this scope
                 _p.val[0] = (uint16x4_t)(vcvt_f16_f32(vld1q_f32(k0)));
                                                                    ^
In file included from /home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_arm.cpp:32:0:
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_gemm_fp16s.h: In function 'void ncnn::innerproduct_gemm_fp16s_neon(const ncnn::Mat&, ncnn::Mat&, const ncnn::Mat&, const ncnn::Mat&, int, const ncnn::Mat&, const ncnn::Option&)':
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_gemm_fp16s.h:123:52: error: 'float16x4_t' was not declared in this scope
                     float32x4_t _w = vcvt_f32_f16((float16x4_t)(vld1_u16(kptr)));
                                                    ^~~~~~~~~~~
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_gemm_fp16s.h:123:80: error: 'vcvt_f32_f16' was not declared in this scope
                     float32x4_t _w = vcvt_f32_f16((float16x4_t)(vld1_u16(kptr)));
                                                                                ^
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_gemm_fp16s.h:217:53: error: 'float16x4_t' was not declared in this scope
                     float32x4_t _w0 = vcvt_f32_f16((float16x4_t)(vget_low_u16(_w01)));
                                                     ^~~~~~~~~~~
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_gemm_fp16s.h:217:85: error: 'vcvt_f32_f16' was not declared in this scope
                     float32x4_t _w0 = vcvt_f32_f16((float16x4_t)(vget_low_u16(_w01)));
                                                                                     ^
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_gemm_fp16s.h:245:52: error: 'float16x4_t' was not declared in this scope
                     float32x4_t _w = vcvt_f32_f16((float16x4_t)(vld1_u16(kptr)));
                                                    ^~~~~~~~~~~
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_gemm_fp16s.h:245:80: error: 'vcvt_f32_f16' was not declared in this scope
                     float32x4_t _w = vcvt_f32_f16((float16x4_t)(vld1_u16(kptr)));
                                                                                ^
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_gemm_fp16s.h:320:52: error: 'float16x4_t' was not declared in this scope
                     float32x4_t _w = vcvt_f32_f16((float16x4_t)(vld1_u16(kptr)));
                                                    ^~~~~~~~~~~
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_gemm_fp16s.h:320:80: error: 'vcvt_f32_f16' was not declared in this scope
                     float32x4_t _w = vcvt_f32_f16((float16x4_t)(vld1_u16(kptr)));
                                                                                ^
[ 42%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/sigmoid_arm.cpp.o
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_gemm_fp16s.h:417:53: error: 'float16x4_t' was not declared in this scope
                     float32x4_t _w0 = vcvt_f32_f16((float16x4_t)(vget_low_u16(_w01)));
                                                     ^~~~~~~~~~~
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_gemm_fp16s.h:417:85: error: 'vcvt_f32_f16' was not declared in this scope
                     float32x4_t _w0 = vcvt_f32_f16((float16x4_t)(vget_low_u16(_w01)));
                                                                                     ^
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_gemm_fp16s.h:436:52: error: 'float16x4_t' was not declared in this scope
                     float32x4_t _w = vcvt_f32_f16((float16x4_t)(vld1_u16(kptr)));
                                                    ^~~~~~~~~~~
/home/guanchao/wangyn/ai/ncnn/src/layer/arm/innerproduct_gemm_fp16s.h:436:80: error: 'vcvt_f32_f16' was not declared in this scope
                     float32x4_t _w = vcvt_f32_f16((float16x4_t)(vld1_u16(kptr)));
                                                                                ^
[ 43%] Building CXX object src/CMakeFiles/ncnn.dir/layer/slice.cpp.o
[ 43%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/slice_arm.cpp.o
[ 44%] Building CXX object src/CMakeFiles/ncnn.dir/layer/softmax.cpp.o
[ 45%] Building CXX object src/CMakeFiles/ncnn.dir/layer/split.cpp.o
[ 45%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/softmax_arm.cpp.o
[ 46%] Building CXX object src/CMakeFiles/ncnn.dir/layer/tanh.cpp.o
[ 47%] Building CXX object src/CMakeFiles/ncnn.dir/layer/threshold.cpp.o
[ 47%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/tanh_arm.cpp.o
[ 47%] Building CXX object src/CMakeFiles/ncnn.dir/layer/tile.cpp.o
[ 48%] Building CXX object src/CMakeFiles/ncnn.dir/layer/rnn.cpp.o
[ 49%] Building CXX object src/CMakeFiles/ncnn.dir/layer/lstm.cpp.o
[ 49%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/rnn_arm.cpp.o
[ 50%] Building CXX object src/CMakeFiles/ncnn.dir/layer/arm/lstm_arm.cpp.o
make[2]: *** [src/CMakeFiles/ncnn.dir/build.make:762: src/CMakeFiles/ncnn.dir/layer/arm/innerproduct_arm.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 50%] Building CXX object src/CMakeFiles/ncnn.dir/layer/binaryop.cpp.o
make[1]: *** [CMakeFiles/Makefile2:143: src/CMakeFiles/ncnn.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
## context | 编译/运行环境 | バックグラウンド
$ cmake --version
cmake version 3.25.0

CMake suite maintained and supported by Kitware (kitware.com/cmake).

$ make --version
GNU Make 4.2.1
Built for x86_64-pc-linux-gnu
Copyright (C) 1988-2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
## how to reproduce | 复现步骤 | 再現方法
1.  cmake -DCMAKE_TOOLCHAIN_FILE=../toolchains/himix200.toolchain.cmake .. 
2.  make -j$(nproc)

**Comment by live106** [comment]
I'm ok with tag 20211208, but not ok for new ones
